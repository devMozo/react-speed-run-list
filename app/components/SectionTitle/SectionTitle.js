import './SectionTitle.scss';
import PropTypes from 'prop-types';

const SectionTitle = ({ title }) => (
  <div className="SectionTitle">
    <h2>{title}</h2>
  </div>
);

SectionTitle.propTypes = {
  title: PropTypes.string,
};

export default SectionTitle;
