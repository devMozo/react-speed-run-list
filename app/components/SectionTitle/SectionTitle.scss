@import '../../fonts/index.scss';

.SectionTitle {
  width: 100%;
  text-align: center;
  font-family: $fontMedium;
  font-size: 30px;
  color: #333;

  h2 {
    margin-bottom: 0 !important;
  }

  &:after {
    content: '';
    width: 150px;
    height: 5px;
    background-color: #333;
    display: inline-block;
  }
}
