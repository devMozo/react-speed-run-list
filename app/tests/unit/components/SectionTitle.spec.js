import React from 'react';
import { shallow } from 'enzyme';
import SectionTitle from '../../../components/SectionTitle/SectionTitle';

describe('<SectionTitle />', () => {
  const MOCKED_PROPS = () => ({
    title: 'Mocked Title',
  });

  it('should render correctly', () => {
    const component = shallow(<SectionTitle {...MOCKED_PROPS()} />);

    expect(component).toMatchSnapshot();
  });

  it('it must contain the correct fields', () => {
    const props = MOCKED_PROPS();
    const component = shallow(<SectionTitle {...props} />);
    const title = component.find('h2');

    expect(title.text()).toEqual(props.title);
  });
});
