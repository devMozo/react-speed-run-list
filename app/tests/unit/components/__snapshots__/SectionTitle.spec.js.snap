// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionTitle /> should render correctly 1`] = `
<div
  className="SectionTitle"
>
  <h2>
    Mocked Title
  </h2>
</div>
`;
